src/utils/useLocationUpdateEffect.ts

```diff
@@ -1,188 +0,0 @@
-import firebase, { UserInfo } from "firebase/app";
-
-import { AnyRoom } from "types/rooms";
-import { AnyVenue, VenueEvent } from "types/venues";
-import { User } from "types/User";
-
-import { updateUserProfile } from "pages/Account/helpers";
-import { useInterval } from "hooks/useInterval";
-
-import { WithId } from "./id";
-import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
-
-const LOCATION_INCREMENT_SECONDS = 10;
-const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
-
-export const updateLocationData = (
-  user: UserInfo,
-  locationName: { [key: string]: number },
-  lastSeenIn: { [key: string]: number } | undefined
-) => {
-  const location = locationName ?? {};
-  const prevRoomName =
-    lastSeenIn &&
-    Object.keys(lastSeenIn).find((lastSeen) => lastSeen.includes("/"));
-
-  if (lastSeenIn && prevRoomName) {
-    delete lastSeenIn[prevRoomName];
-  }
-
-  const roomVenue =
-    locationName && Object.keys(locationName).length
-      ? Object.keys(locationName)[0]
-      : null;
-
-  updateUserProfile(user.uid, {
-    lastSeenAt: getCurrentTimeInMilliseconds(),
-    lastSeenIn:
-      !locationName && !lastSeenIn
-        ? null
-        : lastSeenIn
-        ? { ...lastSeenIn, ...location }
-        : location,
-    room: !locationName && !lastSeenIn ? null : roomVenue,
-  });
-};
-
-// get Profile from the firebase
-// @debt rename this trackLocationEntered?
-export const trackRoomEntered = (
-  user: UserInfo,
-  locationName: { [key: string]: number },
-  lastSeenIn: { [key: string]: number } | undefined
-) => {
-  updateLocationData(user, locationName, lastSeenIn);
-};
-
-export interface TrackRoomEnteredNGProps {
-  user?: UserInfo;
-  venue: AnyVenue;
-  room: AnyRoom;
-  lastSeenIn?: Record<string, number>;
-}
-
-export const trackRoomEnteredNG = ({
-  user,
-  venue,
-  room,
-  lastSeenIn,
-}: TrackRoomEnteredNGProps) => {
-  if (!user) return;
-
-  trackRoomEntered(
-    user,
-    { [`${venue.name}/${room.title}`]: getCurrentTimeInMilliseconds() },
-    lastSeenIn
-  );
-};
```

Comment:
Why was this removed?

Reply:
I don't know how to answer questions like this. It was not used anymore\the alternatives are used instead.

Reply:
These functions served a purpose that I don't believe is handled by any of the ones we have now. Namely the thing I have called out in a few of your PRs/similar related to this, around encoding the `venueName/roomName` slug and similar from the 'base values (venue, room, etc), so that we aren't forced to construct the 'implementation detail' of the format that counting uses at every single place we want to update it.

You raised concerns around violating 'single purpose' functions at that point, at which I suggested that helper functions that could construct the appropriate 'slugs' would also satisfy my need. Yet I haven't seen that in any of the code I have reviewed around this feature/the changes that have been made to it recently.

These functions satisfied that need (though seemingly weren't consistently used everywhere).